numpy/lib/function_base.py

```diff
@@ -3612,6 +3666,78 @@ def quantile(a, q, axis=None, out=None,
     the median if ``q=0.5``, the same as the minimum if ``q=0.0`` and the
     same as the maximum if ``q=1.0``.
 
+    In the simplest case, where all weights are integers,
+    the `weights` argument can be seen as repeating the data in the sample.
+    For example, if a = [1,2,3] and weights = [1,2,3], this will be identical
+    to the case where there are no weights and a = [1,2,2,3,3,3], with the
+    quantiles interpolated in the cumulative probability space of the
+    expanded array.
+
+    The algorithm is illustrated here:
+
+    >>> a = np.array([3, 5, 4])  # sorted
+    >>> q = [0.25, 0.5, 0.75]
+    >>> axis = 0
+    >>> weights = [1, 3, 2]
+
+    Upon expansion, and converting q to quantile space ([0, 1])::
+
+        value       3   4   4   4   5   5
+        cum_prob    0  .2  .4  .6  .8   1
```

Comment:
Which interpolation mode are you showing here?

Reply:
`linear`.  I'll add that to the doc.